src/Essentials/src/FilePicker/FilePicker.ios.cs

```diff
@@ -31,23 +31,29 @@ async Task<IEnumerable<FileResult>> PlatformPickAsync(PickOptions options, bool
 			using var documentPicker = new UIDocumentPickerViewController(allowedUtis, UIDocumentPickerMode.Open);
 #pragma warning restore CA1422 // Validate platform compatibility
 #pragma warning restore CA1416 // Constructor UIDocumentPickerViewController  has [UnsupportedOSPlatform("ios14.0")]
-			if (OperatingSystem.IsIOSVersionAtLeast(11, 0))
-				documentPicker.AllowsMultipleSelection = allowMultiple;
-			documentPicker.Delegate = new PickerDelegate
+			documentPicker.AllowsMultipleSelection = allowMultiple;
+			
+			if (OperatingSystem.IsIOSVersionAtLeast(11) || OperatingSystem.IsMacCatalystVersionAtLeast(13, 1))
 			{
-				PickHandler = urls => GetFileResults(urls, tcs)
-			};
+				documentPicker.DidPickDocumentAtUrls += (_, e) => GetFileResults(e.Urls, tcs);
+			}
+			else 
+			{
+				documentPicker.Delegate = new PickerDelegate
+				{
+					PickHandler = urls => GetFileResults(urls, tcs)
+				};
 
 #if !MACCATALYST
-			if (documentPicker.PresentationController != null)
-			{
-				documentPicker.PresentationController.Delegate =
-					new UIPresentationControllerDelegate(() => GetFileResults(null, tcs));
-			}
+				if (documentPicker.PresentationController != null)
```

Comment:
Here you need to add: `!(OperatingSystem.IsIOSVersionAtLeast(14, 0) && NSProcessInfo.ProcessInfo.IsiOSApplicationOnMac)`, because if not we are back the problem my PR tried to address.

See: https://github.com/dotnet/maui/pull/25668/files